docs/docsite/rst/contributing/creating_collections.rst

```diff
@@ -0,0 +1,138 @@
+    ├── galaxy.yml
+    ├── plugins/
+    │   ├── modules/
+    │   │   └── module1.py
+    │   ├── inventory/
+    │   └── .../
+    └── roles/
+        ├── role1/
+        └── role2
+
+
+``galaxy.yml`` requires the following fields in order to be accepted by galaxy:
+
+.. code-block:: yaml
+
+    namespace: "namespace_name"
+    name: "collection_name"
+    version: "1.0.12"
+    readme: "README.md"
+    authors:
+        - "Author1"
+        - "Author2 (http://author2.example.com)"
+        - "Author3 <author3@example.com>"
+    dependencies:
+        - "collection.name1"
+        - "collection.name2"
+    description: "Example collection metadata"
+    license:
+        - "MIT"
+    repository: "https://www.github.com/my_org/my_collection"
+    documentation: "http://my-docs.example.com"
+    homepage: "http://www.example.com"
+    issues: "https://www.github.com/my_org/my_collection/issues"
+
+
+Required Fields:
+    - ``namespace`` and ``name`` can only contain alphanumeric characters and underscores.
+      Additionally neither can start with underscores or numbers and cannot contain consecutive
+      underscores.
+    - ``version`` numbers must be compatibly with semantic versioning.
+    - ``readme`` contains the filename for the readme file, which can either be markdown (.md) or
+      reStructuredText (.rst).
+
+
+Optional Fields:
+    - ``dependencies``: A list of collections that this collection depends on.
+    - ``description``: A short summary description of the collection.
+    - ``license``: Either a single license or a list of licenses for content inside of a collection.
+      Galaxy currently only accepts SPDX licenses.
+    - ``tags``: a list of tags. These have the same character requirements as ``namespace`` and ``name``.
+    - ``repository``: URL of originating SCM repository.
+    - ``documentation``: URL for online docs.
+    - ``homepage``: URL for project homepage.
+    - ``issues``: URL for issue tracker.
+
+
+Building and Distributing
+=========================
+
+Building collections requires using the ``mazer`` command line tool available at the `Ansible
+Mazer project <https://github.com/ansible/mazer>`_.
+
+Collections are built by running ``mazer build`` from inside the collection's root directory.
+This will create a ``releases`` directory inside the collection with the build artifacts,
+which can be uploaded to Galaxy.
+
+::
+
+    collection/
+    ├── ...
+    ├── releases
+    │   └── namespace_name-collection_name-1.0.12.tar.gz
+    └── ...
+
+    .. note::
+
+        Changing the filename of the tarball in the release directory so that it doesn't match
+        the data in ``galaxy.yml`` will cause the import to fail.
+
+
+Upload Using Mazer
+    Artifacts can be uploaded with mazer using ``mazer publish --api-key=SECRET path/to/namespace_name-collection_name-1.0.12.tar.gz``
+
+    Your API key can be found at `galaxy.ansible.com/me/preferences <https://galaxy.ansible.com/me/preferences>`_.
```

Comment:
I am still _extremely_ of the opinion that this authentication choice is a terrible user experience we should correct. Anyone coming from `ansible-galaxy` and repositories is going to be bewildered at the lack of a login command.

This is still accurate documentation for the time being.